vllm/engine/llm_engine.py

```diff
@@ -348,6 +348,13 @@ def __init__(
                 "vllm.llm_engine",
                 self.observability_config.otlp_traces_endpoint)
 
+        tokenizer_group = self.get_tokenizer_group()
+
+        # Ensure that the function doesn't contain a reference to self,
+        # to avoid engine GC issues
+        def get_tokenizer_for_seq(sequence: Sequence) -> AnyTokenizer:
+            return tokenizer_group.get_lora_tokenizer(sequence.lora_request)
+
```

Comment:
I remember @njhill found this in another pr. why does it occur in this pr?

Reply:
Yes, as mentioned in the description, that part of code is borrowed from Nick's PR.